python312Packages.pwlf: 2.4.0 -> 2.5.1

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
python312Packages.pwlf: 2.4.0 -> 2.5.0

Diff: https://github.com/cjekel/piecewise_linear_fit_py/compare/refs/tags/v2.4.0...v2.5.0

Changelog: https://github.com/cjekel/piecewise_linear_fit_py/blob/refs/tags/v2.5.0/CHANGELOG.md

doronbehar committed:
python312Packages.pwlf: no with lib; in meta

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384594`

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pwlf</li>
    <li>python312Packages.pwlf.dist</li>
    <li>python313Packages.pwlf</li>
    <li>python313Packages.pwlf.dist</li>
  </ul>
</details>

ethancedwards8 commented:
```
>   variance = np.dot(e, e) / (ny - nb)
       > /private/tmp/nix-build-python3.12-pwlf-2.5.0.drv-0/source/pwlf/pwlf.py:1471: RuntimeWarning: invalid value encountered in multiply
       >   pre_var = variance * np.dot(
       > .................
       > ======================================================================
       > ERROR: test_float128 (tests.tests.TestEverything.test_float128)
       > ----------------------------------------------------------------------
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-python3.12-pwlf-2.5.0.drv-0/source/tests/tests.py", line 583, in test_float128
       >     np.linspace(0, 10, 3, dtype=np.float128),
       >                                 ^^^^^^^^^^^
       >   File "/nix/store/1qdjbsbqq9vlbn9g0amydwi4ldmwkn9m-python3.12-numpy-2.2.2/lib/python3.12/site-packages/numpy/__init__.py", line 414, in __getattr__
       >     raise AttributeError("module {!r} has no attribute "
       > AttributeError: module 'numpy' has no attribute 'float128'. Did you mean: 'float16'?
       >
       > ----------------------------------------------------------------------
       > Ran 62 tests in 12.332s
       >
       > FAILED (errors=1)
       For full logs, run 'nix log /nix/store/lpsk8my7sa2crl0zaz16d3f9p200cnh5-python3.12-pwlf-2.5.0.drv'.
error: builder for '/nix/store/n8r8z43xn27gzzsyyhs71b7i8xsdk7rk-python3.13-pwlf-2.5.0.drv' failed with exit code 1;
       last 25 log lines:
       >                        ...
       >                        [ 2.286e-01  7.723e-01]
       >                        [ 2.288e-01  7.703e-01]]
       >  population_energies: [ 8.873e-01  8.894e-01 ...  8.884e-01  8.884e-01]
       > ....../private/tmp/nix-build-python3.13-pwlf-2.5.0.drv-0/source/pwlf/pwlf.py:1461: RuntimeWarning: divide by zero encountered in scalar divide
       >   variance = np.dot(e, e) / (ny - nb)
       > /private/tmp/nix-build-python3.13-pwlf-2.5.0.drv-0/source/pwlf/pwlf.py:1471: RuntimeWarning: invalid value encountered in multiply
       >   pre_var = variance * np.dot(
       > .................
       > ======================================================================
       > ERROR: test_float128 (tests.tests.TestEverything.test_float128)
       > ----------------------------------------------------------------------
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-python3.13-pwlf-2.5.0.drv-0/source/tests/tests.py", line 583, in test_float128
       >     np.linspace(0, 10, 3, dtype=np.float128),
       >                                 ^^^^^^^^^^^
       >   File "/nix/store/km2m3g1ld3ifc3rzmsb2sbimn7qgwq8h-python3.13-numpy-2.2.2/lib/python3.13/site-packages/numpy/__init__.py", line 414, in __getattr__
       >     raise AttributeError("module {!r} has no attribute "
       >                          "{!r}".format(__name__, attr))
       > AttributeError: module 'numpy' has no attribute 'float128'. Did you mean: 'float16'?
       >
       > ----------------------------------------------------------------------
       > Ran 62 tests in 12.737s
       >
       > FAILED (errors=1)
```

doronbehar commented:
Thanks for the report, I reported this upstream:

- https://github.com/cjekel/piecewise_linear_fit_py/issues/126

If they won't patch this quickly, I'll disable these tests.

doronbehar committed:
python312Packages.pwlf: 2.5.0 -> 2.5.1

Fixes upstream failure reported here:

https://github.com/NixOS/nixpkgs/pull/384594#issuecomment-2677114414

Diff: https://github.com/cjekel/piecewise_linear_fit_py/compare/refs/tags/v2.5.0...v2.5.1

Changelog: https://github.com/cjekel/piecewise_linear_fit_py/blob/refs/tags/v2.5.1/CHANGELOG.md

doronbehar commented:
Tested it builds on aarch64-darwin on the Nix community builder. Will merge in a few minutes when CI is fully green.
